authnid/crl/util.py

```diff
@@ -30,25 +29,44 @@ def crl_list_from_disa_html(html):
     return parser.crl_list
 
 
-def write_crls(out_dir, crl_list):
-    for crl_location in crl_list:
-        name = re.split("/", crl_location)[-1]
-        crl = os.path.join(out_dir, name)
-        try:
-            with requests.get(crl_location, stream=True) as r:
-                with open(crl, "wb") as crl_file:
-                    for chunk in r.iter_content(chunk_size=1024):
-                        if chunk:
-                            crl_file.write(chunk)
-        except requests.exceptions.ChunkedEncodingError:
-            print("Error downloading {}, continuing anyway".format(crl_location))
+def write_crl(out_dir, crl_location):
+    name = re.split("/", crl_location)[-1]
+    crl = os.path.join(out_dir, name)
+    with requests.get(crl_location, stream=True) as r:
+        with open(crl, "wb") as crl_file:
+            for chunk in r.iter_content(chunk_size=1024):
+                if chunk:
+                    crl_file.write(chunk)
```

Comment:
This is more of a general question, not this PR, but: since we're writing out the chunks as we get them, if the download fails, we'll still write out a partial CRL. Does that produce a corrupted CRL that the app later ignores or a partial CRL that is handled like the others? Should we remove the file if we receive a chunked encoding error?

Reply:
Yeah, I've been thinking about that. We need to add a retry mechanism for the download (with a max tries) and figure out how to preserve the original, intact CRL if the download genuinely fails. It would probably be corrupt if we write a partial one. [PT icebox story](https://www.pivotaltracker.com/story/show/159075173). The simplest thing to do would probably be to have the bash script copy each CRL from `crl-tmp` to `crl` _only_ if the CRL is the same size or larger.